src/libs/Navigation/Navigation.js

```diff
@@ -78,10 +79,14 @@ function closeDrawer() {
  * @returns {String}
  */
 function getDefaultDrawerState(isSmallScreenWidth) {
-    if (didTapNotificationBeforeReady) {
-        return 'closed';
+    if (didTapNotificationBeforeReady || !isSmallScreenWidth) {
+        return CONST.DRAWER_STATUS.CLOSED;
     }
-    return isSmallScreenWidth ? 'open' : 'closed';
+
+    // When the user opens the app via Report url, close the drawer else open the drawer.
+    const path = getPathFromState(navigationRef.current.getState(), linkingConfig.config).substring(1);
+    // eslint-disable-next-line no-use-before-define
+    return path === ROUTES.HOME || !isDrawerRoute(path) ? CONST.DRAWER_STATUS.OPEN : CONST.DRAWER_STATUS.CLOSED;
```

Comment:
Well! I am not very happy about this change. It has a side effect. Using path to determine default status will only work if we only call this method once on app load which is currently happening. But that was done to fix an issue (Comments on BaseDrawernavigation component). 

A more correct of doing this would be using linkingConfig which is used via react-navigation to configure the app for initialLoad.

Reply:
Are you suggesting that we should set the drawer status on linkingConfig file? I couldn't find the doc that says we can set the drawer state. Could you point me out the link?

Reply:
> Well! I am not very happy about this change. It has a side effect. Using path to determine default status will only work if we only call this method once on app load which is currently happening.

@parasharrajat  I don't get this part. We have discussed the need to simplify the solution and make the URL a source of truth. For that context, I don't know what solution we should do.